woocommerce/woocommerce-android
# Shipping label cost

Closes: #12998
### Description
This PR connects the shipment cost information with the UI. The changes add some missing fields to the UI model and update the UI with the shipping cost information. The changes here focus only on UI changes.

### Testing information
1. Open the app
2. Navigate to the Orders section
3. Open an order eligible for shipping label creation
4. Tap on create shipping label
5. Check that the purchase button is hidden
6. Tap on select a package
7. Select any package
8. Check that after package selection and rates loading, the purchase section is displayed at the bottom of the screen 
9. Check that while there is no rate selected, the purchase button is disabled
10. Select a shipping rate
11. Check that if a shipping rate is selected, the Purchase Label button displays the accurate amount
12. Select a label with signature required options with fee and check that the Purchase Label button displays the accurate amount
13. Expand the Shipment details section
14. Check that the shipment cost section accurately displays the total cost and the subtotal details

### The tests that have been performed
1. Checked that the purchase section displays accurate information
2. Checked that refreshing the shipping rates unselects a package
3. Checked that when no rate is selected, no shipment cost is displayed

### Images/gif

https://github.com/user-attachments/assets/a80cddb9-af57-41db-b9d9-566dcd44e03f



## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
